variables.tf

```diff
@@ -465,4 +465,42 @@ variable "expected_bucket_owner" {
     Account ID of the expected bucket owner. 
     More information: https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucket-owner-condition.html
   EOT
-}
\ No newline at end of file
+}
+variable "event_notification_details" {
+  type = object({
+    enabled = bool
+    lambda_list = optional(list(object({
+      lambda_function_arn = string
+      events              = optional(list(string), ["s3:ObjectCreated:*"])
+      filter_prefix       = string
+      filter_suffix       = string
+    })), [])
+
+    queue_list = optional(list(object({
+      queue_arn = string
+      events    = optional(list(string), ["s3:ObjectCreated:*"])
+    })), [])
+
+    topic_list = optional(list(object({
+      topic_arn = string
+      events    = optional(list(string), ["s3:ObjectCreated:*"])
+    })), [])
+
+  })
+  description = "(optional) S3 event notification details"
+  default = {
+    enabled = false
+  }
+}
+
+variable "create_s3_directory_bucket" {
```

Comment:
Cloudposse has a convention of using a suffix for _enabled for booleans instead of the create_ prefix (which is usually the terraform-aws-modules org convention)

Reply:
Changes done @nitrocode 
https://github.com/cloudposse/terraform-aws-s3-bucket/pull/258